eclair-core/src/test/scala/fr/acinq/eclair/FeaturesSpec.scala

```diff
@@ -105,47 +102,37 @@ class FeaturesSpec extends AnyFunSuite {
   }
 
   test("features compatibility") {
-    assert(areSupported(Features(Set(ActivatedFeature(InitialRoutingSync, Optional)))))
-    assert(areSupported(Features(Set(ActivatedFeature(OptionDataLossProtect, Mandatory)))))
-    assert(areSupported(Features(Set(ActivatedFeature(OptionDataLossProtect, Optional)))))
-    assert(areSupported(Features(Set(ActivatedFeature(ChannelRangeQueries, Mandatory)))))
-    assert(areSupported(Features(Set(ActivatedFeature(ChannelRangeQueries, Optional)))))
-    assert(areSupported(Features(Set(ActivatedFeature(ChannelRangeQueriesExtended, Mandatory)))))
-    assert(areSupported(Features(Set(ActivatedFeature(ChannelRangeQueriesExtended, Optional)))))
-    assert(areSupported(Features(Set(ActivatedFeature(VariableLengthOnion, Mandatory)))))
-    assert(areSupported(Features(Set(ActivatedFeature(VariableLengthOnion, Optional)))))
-    assert(areSupported(Features(Set(ActivatedFeature(PaymentSecret, Mandatory)))))
-    assert(areSupported(Features(Set(ActivatedFeature(PaymentSecret, Optional)))))
-    assert(areSupported(Features(Set(ActivatedFeature(BasicMultiPartPayment, Mandatory)))))
-    assert(areSupported(Features(Set(ActivatedFeature(BasicMultiPartPayment, Optional)))))
-    assert(areSupported(Features(Set(ActivatedFeature(Wumbo, Mandatory)))))
-    assert(areSupported(Features(Set(ActivatedFeature(Wumbo, Optional)))))
-    assert(!areSupported(Features(Set(ActivatedFeature(AnchorOutputs, Mandatory))))) // NB: we're not ready to fully support anchor outputs
-    assert(areSupported(Features(Set(ActivatedFeature(AnchorOutputs, Optional)))))
-
-    val testCases = Map(
-      bin"            00000000000000001011" -> true,
-      bin"            00010000100001000000" -> true,
-      bin"            00100000100000100000" -> true,
-      bin"            00010100000000001000" -> true,
-      bin"            00011000001000000000" -> true,
-      bin"            00101000000000000000" -> true,
-      bin"            00000000010001000000" -> true,
-      bin"            01000000000000000000" -> true,
-      bin"            10000000000000000000" -> true,
-      // unknown optional feature bits
-      bin"        001000000000000000000000" -> true,
-      bin"        100000000000000000000000" -> true,
-      // those are useful for nonreg testing of the areSupported method (which needs to be updated with every new supported mandatory bit)
-      bin"        000100000000000000000000" -> false,
-      bin"        010000000000000000000000" -> false,
-      bin"    0001000000000000000000000000" -> false,
-      bin"    0100000000000000000000000000" -> false,
-      bin"00010000000000000000000000000000" -> false,
-      bin"01000000000000000000000000000000" -> false
+    case class TestCase(ours: Features, theirs: Features, compatible: Boolean)
```

Comment:
Would it be worth it to also check the `areSupported`?
```scala
case class TestCase(ours: Features, theirs: Features, oursSupportTheirs: Boolean, theirSupportOurs: Boolean, compatible: Boolean)
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/1576/commits/c026fa00a7c9e2223760850eae220f810829067b